src/tools/genpolicy/src/registry.rs

```diff
@@ -453,7 +557,23 @@ pub fn get_verity_hash_value(path: &Path) -> Result<String> {
     let hash = verity::traverse_file(&mut file, 0, false, v, &mut verity::no_write)?;
     let result = format!("{:x}", hash);
 
-    Ok(result)
+    file.seek(std::io::SeekFrom::Start(0))?;
+
+    let mut passwd = String::new();
+    for entry_wrap in tar::Archive::new(file).entries()? {
+        let mut entry = entry_wrap?;
+        let entry_path = entry.header().path()?;
```

Comment:
Do we really want to return early if these fail, or rather continue the loop?

Reply:
This kind of error suggests to me that we're dealing with a corrupted tar file. So, returning early seems reasonable to me - at least until/unless we'll find examples that need workarounds.

Reply:
I checked the [docs](https://docs.rs/tar/latest/tar/struct.Header.html#method.path), should be fine to fail here:

> This method may fail if the pathname is not valid Unicode and this is called on a Windows platform.